satip/utils.py

```diff
@@ -211,17 +224,20 @@ def convert_scene_to_dataarray(scene: Scene, band: str, area: str) -> xr.DataArr
     dataset: xr.Dataset = scene.to_xarray_dataset()
     dataarray = dataset.to_array()
 
-    # Assign x_osgb and y_osgb and set some attributes
-    dataarray = dataarray.assign_coords(
-        x_osgb=(("y", "x"), np.float32(osgb_x)),
-        y_osgb=(("y", "x"), np.float32(osgb_y)),
-    )
-    for name in ["x_osgb", "y_osgb"]:
-        dataarray[name].attrs = {"units": "meter", "coordinate_reference_system": "OSGB"}
+    if calculate_osgb:
+        # Assign x_osgb and y_osgb and set some attributes
```

Comment:
I'd maybe suggest moving lines 217 & 218 to here, so you only need one `if calculate_osgb` block :slightly_smiling_face:

Reply:
Sounds good!